lib/main.dart

```diff
@@ -51,9 +51,17 @@ void main(List<String> args) async {
       version =
           raw ? 'latest' : (await latestPublishedVersion(channel, flavor));
     } else if (sdk == 'main') {
-      channel = 'be';
-      version =
-          raw ? 'latest' : (await latestPublishedVersion(channel, flavor));
+      // TODO(b/299435467): Temporary forward compatibility that
+      // automatically migrates users when the be channel is renamed to main.
+      try {
+        channel = 'main';
+        version =
```

Comment:
Note that the `latestPublishedVersion` method won't be called for all paths here, like if the `raw` flavor had been specified. I take it we're trying to validate that a url exists for the `main` channel, and if no, fallback on the `be` channel? If so we'd need to validate the urls for the raw flavor as well.

Will this be a very quick transition? Using `be` is I think a pretty specialized use case. We could do something like start double-publishing to main and be now, have the (current) 1.5.0 version of this action read from `be`, and publish a newer 1.6.0 version which only read from `main`.

And I know there are some used cases for having a `raw` flavor, but they're so rarified I wonder if we couldn't remove that as well.

Reply:
Ah I missed the raw variable.

Only the raw flavor makes sense for main/be and only the latest version, since we don't publish or sign the main channel. I updated the temporary logic to handle the supported cases.

I tend to support removing the flavor support as an explicit input and instead default to the raw flavor on main only. I'll clean this up in a follow up change when we've done the migration.

I hope to do the migration this week or the next one, no big rush. Deploying this package may be a limiting factor on how fast we can roll it out? But the main channel isn't really that supported.

The migration will happen as an atomic switch when tools/VERSION in the Dart SDK is modified to the new channel. The rest of the infrastructure will respect the channel name in there and start uploading to the new places. Services like api.dart.dev and setup-dart will handle the switch automatically when the new channel pops into existence. The infra isn't quite set up to upload to both places though we could potentially do that, it just isn't the migration path I've planned out at this time.

Reply:
> Deploying this package may be a limiting factor on how fast we can roll it out?

We can release this at any time - it's just creating a github release (https://github.com/dart-lang/setup-dart/wiki/Publishing-procedure). I can take care of that after your work lands.